pkg/beholder/httpclient.go

```diff
@@ -0,0 +1,203 @@
+package beholder
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	sdklog "go.opentelemetry.io/otel/sdk/log"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	sdkresource "go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+// Used for testing to override the default exporter
+type otlploghttpFactory func(options ...otlploghttp.Option) (sdklog.Exporter, error)
+
+func newCertFromFile(certFile string) (*x509.CertPool, error) {
+	b, err := os.ReadFile(certFile)
+	if err != nil {
+		return nil, err
+	}
+	cp := x509.NewCertPool()
+	if !cp.AppendCertsFromPEM(b) {
+		return nil, fmt.Errorf("credentials: failed to append certificates")
+	}
+	return cp, nil
+}
+
+func newHTTPClient(cfg Config, otlploghttpNew otlploghttpFactory) (*Client, error) {
+	noop := NewNoopClient()
+	baseResource, err := newOtelResource(cfg)
+	if err != nil {
+		return noop, err
```

Comment:
Why return noop instead of nil? Typically when error is not nil, the rest of the return values are undefined. Can we leave it up to the caller to use a `NewNoopClient()` if they want to?